src/common/vordr.ts

```diff
@@ -18,29 +18,71 @@ export const validateVordrIPs = (ip: string): boolean =>
   isIP(ip) && isInSubnet(ip, vordrIPs);
 
 export const validateVordrWords = (content: string): boolean => {
+  if (!content) {
+    return false;
+  }
+
   const lowerCaseContent = content.toLowerCase();
   return vordrWords.some((word) => lowerCaseContent.includes(word));
 };
 
+type VordrFilterInput = {
+  comment?: Comment;
+  post?: Post | FreeformPost;
+  submission?: Submission;
+};
+
+const getTypeAndId = ({ comment, post, submission }: VordrFilterInput) => {
+  if (comment) {
+    return { type: 'comment', id: comment.id };
+  }
+
+  if (post) {
+    return { type: 'post', id: post.id };
+  }
+
+  if (submission) {
+    return { type: 'submission', id: submission.id };
+  }
+
+  return { type: 'unknown', id: 'unknown' };
+};
+
 export const checkWithVordr = async (
-  comment: Comment,
-  { userId, con, req }: Context,
+  { comment, post, submission }: VordrFilterInput,
+  {
+    userId,
+    con,
+    req,
+  }: {
+    userId: string;
+    con: DataSource | EntityManager;
+    req: Pick<FastifyRequest, 'ip'>;
+  },
 ): Promise<boolean> => {
+  const { type, id } = getTypeAndId({ comment, post, submission });
```

Comment:
can we maybe use instanceof instead of mapping types to strings by hand?

Reply:
How do you envision it to be?

Reply:
Instead of `type === 'comment'` do `comment instanceof Comment`.

Reply:
Also what happens if you pass post and coment to `getTypeAndId` you would always get `type: comment`, so it feels a bit error prone

Reply:
I think better would be to change `checkWithVordr` to accept first argument as entity with generic and then use instanceof inside to decide on type.

Reply:
Refactored to accept as I mentioned above.